lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -653,7 +614,8 @@ export default class BitcoinProcessor {
     for (let transactionIndex = 1; transactionIndex < transactions.length; transactionIndex++) {
       const transaction = transactions[transactionIndex];
 
-      const isSidetreeTransaction = this.isSidetreeTransaction(transaction);
+      const sidetreeData = SidetreeTransactionData.parse(transaction, this.sidetreePrefix);
```

Comment:
This call appears to be quite expensive but we call it twice for every transaction. I wonder if it is worth avoiding this. It is fine to file an issue if you think it is worthwhile but not mainnet blocking.

Reply:
Yea that is a bigger refactor issue and I have opened issue #434 to track this.